### ISSUE TITLE
Support subvolume names with UTF-8 characters

### ISSUE THREAD

**Issue by p1otr** [digint/btrbk#392]
Probably related to #213.

Please add support for UTF-8 in subvolume names

```
# btrfs subvolume show /archiwum/zdjęcia
zdjęcia
        Name:                   zdjęcia
        UUID:                   4b1464ea-542e-5a45-a135-f158a20f4f0c
        Parent UUID:            -
        Received UUID:          -
        Creation time:          2021-04-29 17:57:47 +0200
        Subvolume ID:           257
        Generation:             954
        Gen at creation:        15
        Parent ID:              5
        Top level ID:           5
        Flags:                  -
        Snapshot(s):
# btrbk dryrun
ERROR: Ambiguous file for option "subvolume" in "/etc/btrbk/btrbk.conf" line 113: zdjęcia
```

Thanks for this great tool, BTW

**Comment by digint** [comment]
This has been asked before (#120 and others I can't find right now), and I completely see the need for this for regular users.

As a sysadmin, let me tell you that having non-ascii characters in your file hierarchy is asking for trouble. These days, most systems can cope with utf8, but character encoding is a bitch and there's much more to it...

That said, I got around to work on this, here's what I got so far:

 - I added `--unsafe-filenames` command-line option in bb368cfb217735b7f4f2b3dae074ed54685ab66f, which removes most hard constraints on file names. This patch assumes that external tools (on the host running btrbk as well as remote hosts) can handle the encoding used. I ran some basic tests, and it seems to work.
 - I created the [unsafe-filenames](/digint/btrbk/commits/unsafe-filenames) branch, containing more code to handle whitespaces in file names by quoting them when running external commands.

Please have a look at it, and tell me if it works for you:
```
wget https://raw.githubusercontent.com/digint/btrbk/unsafe-filenames/btrbk
chmod +x btrbk
sudo ./btrbk run -n
```

**Comment by p1otr** [comment]
seems to work fine with `--unsafe-filenames`, thanks!

**Comment by Pascalio** [comment]
Wonderful! It works nicely for me too, thank you!

**Comment by Pascalio** [comment]
The --unsafe-filenames branch has been working perfectly fine for me. As you said, for regular users, it's fundamental to support UTF-8 filenames. Would you consider merging it into the main stream, so that we can track the latest tags, please?

**Comment by digint** [comment]
It's already on the master branch, and ready to be released. The only thing is that I need to do some testing on it before releasing, and I'm super busy at the moment... Try it, should work.

**Comment by digint** [comment]
Included in btrbk-0.32.0, closing issue
